[Core][Flaky Test] FMT - shared_client_tests::test_pipeline_can_reconnect::use_cluster_1_false

### Test Name

shared_client_tests::test_pipeline_can_reconnect::use_cluster_1_false

### Test Location

tests/test_client.rs:1879:18

### Failure Permlink

https://github.com/valkey-io/valkey-glide/actions/runs/18150274381/job/51659810645#step:5:2736

### Frequency

_No response_

### Steps to Reproduce

_No response_

### System Information

ubuntu, linux, ubuntu-24.04, x64, x86_64-unknown-linux-gnu, pypi, npm, maven

### Language and Version

Rust

### Engine Version

redis 7.0

### Logs

2025-10-01T03:41:35.403322Z DEBUG redis::retry_strategies: Starting RetryStrategy with values: RetryStrategy { factor: 100, exponent_base: 2, number_of_retries: 5, jitter_percent: 20 }
2025-10-01T03:41:35.403443Z DEBUG logger_core: connection creation - Attempting connection to Host: `127.0.0.1`, Port: 42555
2025-10-01T03:41:35.403519Z DEBUG redis::aio::connection: Creating TCP connection for node: "127.0.0.1:42555", IP: 127.0.0.1
2025-10-01T03:41:35.404098Z DEBUG logger_core: connection creation - Connection to 127.0.0.1:42555 created
2025-10-01T03:41:35.404331Z  INFO logger_core: Connection configuration - 
Addresses: 127.0.0.1:42555
TLS mode: No TLS
Standalone mode
Read from Replica mode: Only primary
database ID: 0
Protocol: RESP3
2025-10-01T03:41:35.404375Z DEBUG redis::retry_strategies: Starting RetryStrategy with values: RetryStrategy { factor: 100, exponent_base: 2, number_of_retries: 5, jitter_percent: 20 }
2025-10-01T03:41:35.404442Z DEBUG logger_core: connection creation - Attempting connection to Host: `127.0.0.1`, Port: 42555
2025-10-01T03:41:35.404516Z DEBUG redis::aio::connection: Creating TCP connection for node: "127.0.0.1:42555", IP: 127.0.0.1
2025-10-01T03:41:35.405113Z DEBUG logger_core: connection creation - Connection to 127.0.0.1:42555 created
2025-10-01T03:41:35.405575Z DEBUG logger_core: StandaloneClient - connection checker stopped after connection was dropped
2025-10-01T03:41:35.405759Z DEBUG logger_core: StandaloneClient - connection checker has triggered reconnect
2025-10-01T03:41:35.405786Z DEBUG logger_core: reconnect - starting
2025-10-01T03:41:35.405869Z DEBUG redis::aio::connection: Creating TCP connection for node: "127.0.0.1:42555", IP: 127.0.0.1
2025-10-01T03:41:35.405990Z  WARN logger_core: pipeline request - received disconnect error `Failed to receive a response due to a fatal error - FatalReceiveError: channel closed`
test shared_client_tests::test_pipeline_can_reconnect::use_cluster_1_false ... FAILED




failures:

---- shared_client_tests::test_pipeline_can_reconnect::use_cluster_1_false stdout ----

thread 'shared_client_tests::test_pipeline_can_reconnect::use_cluster_1_false' panicked at tests/test_client.rs:1879:18:
Pipeline failed: Failed to receive a response due to a fatal error - FatalReceiveError: channel closed
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
    shared_client_tests::test_pipeline_can_reconnect::use_cluster_1_false

test result: FAILED. 69 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 156.87s

error: test failed, to rerun pass `--test test_client`
Error: Process completed with exit code 101.

### Screenshots

_No response_

### Glide Version

2.1.0

### Expected Behavior

_No response_

### Actual Behavior

_No response_

### Possible Fixes

_No response_

--- a/glide-core/tests/test_client.rs
+++ b/glide-core/tests/test_client.rs
@@ -1998,22 +1998,25 @@ pub(crate) mod shared_client_tests {
             )
             .await;
 
-            let res = retry(|| async {
-                let mut client = test_basics.client.clone();
-                client
-                    .send_pipeline(
-                        &pipeline,
-                        None,
-                        false,
-                        None,
-                        PipelineRetryStrategy {
-                            retry_server_error: true,
-                            retry_connection_error: false,
-                        },
-                    )
-                    .await
-                    .ok()
-            })
+            let res = retry_until_timeout(
+                || async {
+                    let mut client = test_basics.client.clone();
+                    client
+                        .send_pipeline(
+                            &pipeline,
+                            None,
+                            false,
+                            None,
+                            PipelineRetryStrategy {
+                                retry_server_error: true,
+                                retry_connection_error: false,
+                            },
+                        )
+                        .await
+                        .ok()
+                },
+                std::time::Duration::from_millis(10_000),
+            )
             .await;
 
             let expected = Value::Array(vec![
@@ -2152,22 +2155,25 @@ pub(crate) mod shared_client_tests {
             // Kill all connections.
             kill_connection(&mut test_basics.client).await;
 
-            let res = retry(|| async {
-                let mut client = test_basics.client.clone();
-                client
-                    .send_pipeline(
-                        &pipeline,
-                        None,
-                        false,
-                        None,
-                        PipelineRetryStrategy {
-                            retry_server_error: true,
-                            retry_connection_error: false,
-                        },
-                    )
-                    .await
-                    .ok()
-            })
+            let res = retry_until_timeout(
+                || async {
+                    let mut client = test_basics.client.clone();
+                    client
+                        .send_pipeline(
+                            &pipeline,
+                            None,
+                            false,
+                            None,
+                            PipelineRetryStrategy {
+                                retry_server_error: true,
+                                retry_connection_error: false,
+                            },
+                        )
+                        .await
+                        .ok()
+                },
+                std::time::Duration::from_millis(10_000),
+            )
             .await;
 
             let expected = Value::Array(vec![

